workspace_status.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# p_() takes 2 inputs, the name of a variable and a value. It then prints the
+# variable and the value to be fed into bazel's --workspace_status flag. The
+# interesting thing is that it will *only* print the given value if there is
+# *not* an environment variable with the same name that is already set (with
+# some nonempty string). This way, bazel will either read in these ENV vars, but
```

Comment:
I don't follow why you're doing this; in general, bazel rules aren't dependent on the environment, so anything you want as an input to the rule should be set in the workspace status.

Reply:
The missing context here that I forgot to add is that this way, I can pass in substitution vars into the GCB (cloudbuild.yaml), which can then be passed in as env vars to the image that is running in a step (to be consumed by this p_() function into bazel). The idea is that I can now reuse the same cloudbuild.yaml for different targets.

Reply:
I still don't really understand. If I'm understanding the comment and this code properly, if you set e.g. `STABLE_IMG_REGISTRY` in the environment, then it won't be available to bazel rules anymore. Why do you want to do that?

Reply:
Sorry, allow me to reword your example. If you set `STABLE_IMG_REGISTRY` in the environment, then bazel rules *will* use them (as per `p_()` in `workspace_status.sh`). Like you said, bazel rules aren't dependent on the environment; and so I wrote `p_()` to pass along env vars into bazel. The complete picture (accounting for how GCB will use these env vars) is like this:

1. User sets GCB substitution variables, and passes them into the environment for a `step` definition in the GCB YAML. E.g., `gcloud container builds submit --substitutions=STABLE_IMG_NAME=foo`
2. The STABLE_IMG_NAME env var is exposed into the bazel build step inside GCB.
3. The bazel build step's invocation of `workspace_status.sh` then uses `foo` for STABLE_IMG_NAME instead of `cip` in `workspace_status.sh`, because it is set in the environment.
4. Now bazel uses STABLE_IMG_NAME=foo in its build.

Reply:
ohhh, now I understand. I was a bit confused by "only prints the value if the environment variable isn't set" - I thought that meant it wouldn't print anything at all.

It might be clearer to write something like

`p_` takes two arguments to define a bazel workspace status variable:
- the name of the variable
- a default value

If an environment variable with the corresponding name is set, its value is used. Otherwise, the provided default value is used.

Reply:
Done. Looking at this again, your comment for `p_()` is a lot easier to understand, so I used it as-is. I hope you don't mind. =)